core/src/main/scala/org/apache/spark/eventhubs/client/CachedEventHubsReceiver.scala

```diff
@@ -59,11 +55,12 @@ private[spark] trait CachedReceiver {
  *
  * @param ehConf the [[EventHubsConf]] which contains the connection string used to connect to Event Hubs
  * @param nAndP  the Event Hub name and partition that the receiver is connected to.
- */
+*/
 private[client] class CachedEventHubsReceiver private (ehConf: EventHubsConf,
                                                        nAndP: NameAndPartition,
-                                                       startSeqNo: SequenceNumber)
-    extends Logging {
+                                                       startSeqNo: SequenceNumber,
+                                                       eventHubsReceiverListener: Option[MetricPlugin])
```

Comment:
Instead of adding a parameter, let's create a filed for the plugin as the following and remove all unnecessary changes.
  private lazy val metricPlugin: Option[MetricPlugin] = ehConf.metricPlugin()
  private lazy val client: EventHubClient = ClientConnectionPool.borrowClient(ehConf)
  private var receiver: PartitionReceiver = createReceiver(startSeqNo)